app/controllers/playlists_controller.rb

```diff
@@ -42,6 +42,24 @@ def structure
     @logical_order = WithProxyForObject.new(@logical_order, members)
   end
 
+  def around_delete_action
+    # Only allow deleting empty playlists. Avoids difficulty of restoring a
+    # full playlist from preservation and supporting restoring accidentally
+    # deleted playlists.
+    count = if @change_set.resource.is_a? Playlist
```

Comment:
I think maybe you wrote the other one first and then used the same logic, but I read this one first and it was sort of confusing why there's a count variable. I got a bit tripped up by the zero value (didn't see what it had to do with a yield at first) and I think it would be a little easier to understand if line 55 instead were more like `if @change_set.resource.is_a? Playlist && @change_set.resource.member_ids.count.positive?` What do you think?

Also, it's nice to have these comments!

Reply:
good suggestion. 👍